spec/lib/object_serializer_class_methods_spec.rb

```diff
@@ -486,4 +486,63 @@ def year_since_release_calculator(release_year)
       end
     end
   end
+
+  describe '#pluralize_type' do
+    subject(:serializable_hash) { MovieSerializer.new(movie).serializable_hash }
+
+    before do
+      MovieSerializer.pluralize_type pluralize
+    end
+
+    after do
+      MovieSerializer.pluralize_type nil
+      MovieSerializer.set_type :movie
+    end
+
+    context 'when pluralize is true' do
+      let(:pluralize) { true }
+
+      it 'returns correct hash which type equals pluralized value' do
+        expect(serializable_hash[:data][:type]).to eq :movies
+      end
+    end
+
+    context 'when pluralize is false' do
+      let(:pluralize) { false }
+
+      it 'returns correct hash which type equals non-pluralized value' do
+        expect(serializable_hash[:data][:type]).to eq :movie
+      end
+    end
+  end
+
+  describe '#pluralize_type after #set_type' do
```

Comment:
As it stands there's an issue where type is not correctly pluralized in relationships. The following test shows this:

```
    context 'when pluralizing a relationship type after #set_type' do
      before do
        ActorSerializer.pluralize_type true
      end

      after do
        ActorSerializer.pluralize_type nil
      end

      it 'returns correct hash which relationship type equals transformed set_type value' do
        expect(serializable_hash[:data][:relationships][:actors][:data][0][:type]).to eq(:actors)
        expect(serializable_hash[:included][:type]).to eq(:actors)
      end
    end
```

This also means there's a mismatch between relationship and include types. Currently:

```
> serializable_hash[:data][:relationships][:actors][:data][0][:type]
# => :actor
> serializable_hash[:included][:type]
# => :actors
```

Reply:
Unfortunately, as far as I understand, `fast_jsonapi` does not actually "know" internally what serializer is associated with any given relationship. The relationship data is built via simply reflecting on the object and constructing the abbreviated relationship structure. In theory it's possible to make it aware of the serializer, but see my original "Additional Note" -- I ended up being worried about missing performance targets, so I didn't implement a change here.

In order to make this scenario work, I implemented the ability to mark a specific relationship as plural with `pluralize_type: true`. (Check out `object_serializer_pluralization_spec.rb`.) This isn't ideal, but it solves this scenario in lieu of a higher-level decision about perf targets.

Reply:
I sort of glossed over that note when I first reviewed this PR, and after banging my head against this for a bit, I'm coming to the same conclusion... I'm going to investigate this a bit more because it is important for my use case, but I understand that the complexity involved in such a refactor may not be worth the effort.

At any rate it may not be worth including in this PR, as it already achieves the majority of what it sets out to do.